tests/src/Kernel/IndexTest.php

```diff
@@ -84,19 +108,39 @@ protected function queryIndex($docId) {
 
     // Query URI for fetching the document from elasticsearch.
     $uri = 'http://' . $elasticsearch_host . ':9200/simple/_search?q=id:' . $docId;
+    return $this->httpRequest($uri);
+  }
 
-    // Query elasticsearch.
-    // Use Curl for now because http client middleware fails in KernelTests
-    // (See: https://www.drupal.org/project/drupal/issues/2571475)
-    $curl = curl_init();
-    curl_setopt($curl, CURLOPT_URL, $uri);
-    curl_setopt($curl, CURLOPT_HTTPGET, TRUE);
-    curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
-    $json = curl_exec($curl);
+  /**
+   * Test index mapping.
+   */
+  public function testIndexMapping() {
+    $elasticsearch_host = $this
+      ->config('elasticsearch_helper.settings')
+      ->get('elasticsearch_helper.host');
 
-    $response = json_decode($json, TRUE);
+    // Query URI for fetching the document from elasticsearch.
+    $uri = 'http://' . $elasticsearch_host . ':9200/simple/_mapping';
+
+    $response = $this->httpRequest($uri);
+
+    if (ElasticsearchClientVersion::getMajorVersion() >= 7) {
+      // ES7 mapping structure with no type name.
+      $properties = $response['simple']['mappings']['properties'];
+    } else {
+      // ES6 mapping structure with type name.
+      $properties = $response['simple']['mappings']['node']['properties'];
+    }
 
-    return $response;
+    $this->assertEqual($properties['id']['type'], 'text', 'ID field is found');
+    $this->assertEqual($properties['status']['type'], 'boolean', 'Status field is found');
+    $this->assertEqual($properties['title']['type'], 'text', 'Title field is found');
+    $this->assertEqual(
+      $properties['title']['fields'],
+      ['keyword' => ['type' => 'keyword', 'ignore_above' => 256]],
+      'Title sub-field is found'
+    );
+    $this->assertEqual($properties['uuid']['type'], 'text', 'UUID field is found');
```

Comment:
I wonder why `id` and `uuid` test positive for `text` type. Fields `id` and `uuid` are supposed to be keywords (https://github.com/wunderio/elasticsearch_helper/blob/8.x-7.x/tests/modules/elasticsearch_helper_test/src/Plugin/ElasticsearchIndex/SimpleNodeIndex.php#L27).

Reply:
Good catch. I need to find to setup the mapping on each test. The problem now is that the index is deleted on each test run, so the mapping changes.